fontdrasil/src/types.rs

```diff
@@ -156,4 +157,18 @@ impl WidthClass {
             WidthClass::UltraExpanded => 200.0,
         }
     }
+
+    pub fn from_percent(percent: f32) -> Self {
```

Comment:
Should document what happens if pct doesn't exactly match a class. Changing the name might suffice, e.g. `nearest_class(pct: f32)` I'm immediately clear what I get.